src/sample-file.ts

```diff
@@ -0,0 +1,63 @@
+import * as path from 'path';
+import * as fs from 'fs-extra';
+import { Component } from './component';
+import { Project } from './project';
+import { fileExists } from './util';
+
+/**
+ * Options for the SampleFile object.
+ *
+ * @field contents - The contents of the file.
+ */
+export interface SampleFileOptions {
+  /**
+   * The contents of the file to write.
+   */
+  readonly contents: string;
+}
+
+/**
+ * Produces a file with the given contents but only once, if the file doesn't already exist.
+ * Use this for creating example code files or other resources.
+ */
+export class SampleFile extends Component {
+  private readonly filePath: string;
+  private options: SampleFileOptions;
+  public created: boolean = false;
```

Comment:
Don't allow accessing this directly. If there is a use case of a getter, then add a getter.

```suggestion
  private created: boolean = false;
```